src/ComposerJson.php

```diff
@@ -125,6 +144,76 @@ private function extractAutoloadPaths(string $basePath, array $autoload, bool $i
         return $result;
     }
 
+    /**
+     * @param array<string> $exclude
+     * @return array<string, bool>
+     * @throws InvalidPathException
+     */
+    private function extractAutoloadExcludeRegexps(string $basePath, array $exclude, bool $isDev): array
+    {
+        $regexps = [];
+
+        foreach ($exclude as $path) {
+            $regexps[$this->resolveAutoloadExclude($basePath, $path)] = $isDev;
+        }
+
+        return $regexps;
+    }
+
+    /**
+     * Implementation copied from composer/composer.
```

Comment:
This is the main reason why I didnt implement this. 

Current, people can solve this on their own via custom Glob solution (e.g. `Webmozart\Glob\Glob`) within config file:

```php
return $config
    ->addPathsToExclude(Glob::glob(__DIR__ . '/bin/*.php'), true)
```

Let me think about it again if I want such "copy dependency" here...

Reply:
Technically [`exclude-from-classmap` not a glob](https://getcomposer.org/doc/04-schema.md#exclude-files-from-classmaps), only `*` and `**` supported. Custom `addPathsToExclude` will also require duplicating these dependencies (in `composer.json` and `composer-dependency-analyser.php`).

_PS: Seems it is supported by  [composer-require-checker](https://github.com/maglnet/ComposerRequireChecker/releases/tag/0.2.0), but not `composer-unused`_

Reply:
Ok, if you add [enough real-exclusion tests](https://github.com/shipmonk-rnd/composer-dependency-analyser/pull/140#discussion_r1634861869) I think we can merge this.